### PR TITLE
Implement diataxis

### LINKED ISSUE #1534: feature request: implement diataxis
https://diataxis.fr/

### PR DESCRIPTION
**Pull request by Mackenzie-OO7**
closes #1534 

## Summary by CodeRabbit

- **New Features**
	- Enhanced navigation structure for improved user experience.
	- Expanded eligibility criteria and application process for the Celestia Foundation Delegation Program.
	- Added detailed sections on the integration of Arbitrum Nitro with Celestia and Blobstream contracts.

- **Documentation**
	- Updated and clarified various how-to guides for better accuracy and usability, including links for deployment and node setup.
	- Improved clarity in documents related to Blobstream integration and proof querying.
	- Streamlined links for staking and governance proposal submissions for easier access.

- **Bug Fixes**
	- Corrected links in multiple guides to ensure users access the correct resources.

### CODE REVIEW

**Comment by jcstein** [comment]
> should i also put related files in sub-folders under each category? say create sub-folders for integrations, running a node, etc. under how-to guides, rather than just have everything in one main folder
let's try to keep the broken links to a minimum here if possible

**Comment by jcstein** [comment]
i'm going to take some time to review this and give you better answers on these points. as i realize this Issue inherently breaks a lot of links

**Comment by jcstein** [comment]
Screenshot 2024-09-30 at 9 19 15 PM
🔥

File: .vitepress/config.ts [hunk 1]
```diff
@@ -8,827 +8,852 @@ const { BASE: base = "/" } = process.env;
...
-                  text: "Astria",
-                  collapsed: true,
-                  items: [
-                    { text: "Documentation", link: "https://docs.astria.org" },
-                    {
-                      text: "Just deploy",
-                      link: "https://docs.astria.org/developer/tutorials/install-the-cli",
-                    },
-                  ],
+            text: "How To Guides",
```
Comment (jcstein) [inline .vitepress/config.ts:378]:
```suggestion
            text: "How-to guides",
```

Comment (jcstein) [reply .vitepress/config.ts:378]: and any other instances of this, should change to this 🙏

File: .vitepress/config.ts [hunk 2]
```diff
@@ -8,827 +8,852 @@ const { BASE: base = "/" } = process.env;
...
-                    {
-                      text: "Run an OP Stack devnet posting Celestia",
-                      link: "/developers/optimism",
-                    },
-                    {
-                      text: "OP Stack testnets",
-                      collapsed: true,
-                      items: [
+                    text: "Build whatever",
+                    link: "/how-to-guides/build-whatever",
```
Comment (jcstein) [inline .vitepress/config.ts:519]: `celestia.org/how-to-guides/build-whatever` is clear and also has a nice ring. thank you @Mackenzie-OO7 and Diataxis!

**Review by jcstein — CHANGES_REQUESTED** [review]
LGTM aside a few small requests!

**Comment by Mackenzie-OO7** [comment]
> i'm going to take some time to review this and give you better answers on these points. as i realize this Issue inherently breaks a lot of links

Thanks for reviewing!

File: index.md [hunk 3]
```diff
@@ -23,13 +23,13 @@ features:
...
-    details: Access the dynamic scaling unlocked by data availability sampling, where scale increases with the number of users.
+  - title: How-to guides
+    details: Explore step-by-step guides for running a node, posting data blobs, building applications and sovereign rollups on Celestia.
     link: /how-to-guides/nodes-overview
     icon: 📈
-  - title: Developers
-    details: Create applications using your favorite VM or define your own. Build sovereign rollups, a new type of self-governing blockchain with minimal platform risk.
-    link: /how-to-guides/build-whatever
+  - title: Tutorials
+    details: Access tutorials for interacting with celestia-node through the node API.
```
Comment (jcstein) [inline index.md:31]:
```suggestion
    details: Access tutorials for interacting with Celestia, starting with celestia-node through the node API.
```

**Comment by jcstein** [comment]
> might be more organized, but yeah it'll break a lot of links

I'm working on scoping out server-level redirects so this won't be an issue!

> Thanks for reviewing!

you're welcome! it was exciting to see.

**Comment by Mackenzie-OO7** [comment]
> here's a table showing the old paths and the new paths that they should redirect to:

i don't think it's necessary, plus now that you outlined it like this, it does look more visually appealing😅

**Comment by jcstein** [comment]
yeah my only real nitpick is that the links would get long!

**Comment by Mackenzie-OO7** [comment]
> yeah my only real nitpick is that the links would get long!

right. any other changes to be made? i think this is the best thing I've worked on so far😁

**Comment by jcstein** [comment]
just waiting to hear back on redirects. agreed, this is awesome!

**Comment by jcstein** [comment]
and we'll get #1603 in once this is all done

**Comment by Mackenzie-OO7** [comment]
> can you please double-check the link changes in the table of my comment above, before we make this live @Mackenzie-OO7 ?

sure! hang on

**Comment by Mackenzie-OO7** [comment]
> can you please double-check the link changes in the table of my comment above, before we make this live @Mackenzie-OO7 ?

all good!

File: learn/staking-governance-supply.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ Learn
 | Delegation support   | Yes            |
 
 Learn how to
-[stake on your own at the community dashboards](../../nodes/staking.md).
+[stake on your own at the community dashboards](../../how-to-guides/staking.md).
```
[the remark this answers is not in the document]
Comment (jcstein) [reply learn/staking-governance-supply.md:30]: this is under learn, right @Mackenzie-OO7 ?

[the remark this answers is not in the document]
Comment (Mackenzie-OO7) [reply learn/staking-governance-supply.md:30]: it is. thanks for spotting that😅
i'll change it now

[the remark this answers is not in the document]
Comment (jcstein) [reply learn/staking-governance-supply.md]: thanks! no worries, coderabbit FTW

**Comment by jcstein** [comment]
can you please give one last review of the table for redirects?

**Review by jcstein — APPROVED** [review]

**Comment by jcstein** [comment]
Final CSV: https://gist.github.com/jcstein/7976777202d155ccf73e0eed234d0aa2

**Pull request merged**